src/interfaces/web3-connection-options.ts

```diff
@@ -1,8 +1,10 @@
+import {HttpProvider, IpcProvider, WebsocketProvider} from 'web3-core';
 import {HttpProviderOptions, WebsocketProviderOptions} from 'web3-core-helpers';
 
 export interface Web3ConnectionOptions {
   web3Host?: string;
   privateKey?: string;
   web3ProviderOptions?: HttpProviderOptions | WebsocketProviderOptions; // you can provide a node server if you're using IPC
+  provider?: HttpProvider|IpcProvider|WebsocketProvider; // connect through a wallet's supplied web3 provider
```

Comment:
We let Web3Connection decide its own provider by using web3host and web3provider options.

If you need to re-use the same web3connection you can do so without having to call "start()", check integration tests for example of multiple uses of the same web3connection with _one_ start.

Reply:
removed as agreed on discord